### PR TITLE
[ll] Port cube example and fix gl queue submission [13/..]

### PR DESCRIPTION
**Pull request by msiglreith**
* `Cube` demo now runs on GL (excluding resizing..)
* Fix queue submission on GL as mentioned on the prior PR, based on the old `Device` interface

(Removed unneeded attribute from the last PR)

### CODE REVIEW

File: examples/cube/main.rs [hunk 1]
```diff
@@ -16,11 +16,13 @@ extern crate cgmath;
 #[macro_use]
 extern crate gfx;
 extern crate gfx_app;
+extern crate gfx_core;
```
Comment (kvark) [inline examples/cube/main.rs]: we need to make it not a required dependency, at the end of the day

File: src/core/src/lib.rs [hunk 2]
```diff
@@ -398,10 +398,22 @@ pub struct QueueSubmit<'a, B: Backend + 'a> {
 /// CommandBuffers will be later submitted to command queues instead of the device.
 pub trait CommandQueue<B: Backend> {
     /// Submit command buffers to queue for execution.
-    unsafe fn submit(&mut self, submit_infos: &[QueueSubmit<B>], fence: Option<&mut <B::Resources as Resources>::Fence>);
+    /// `fence` will be signalled after submission and _must_ be unsignalled.
+    // TODO: `access` legacy (handle API)
+    #[doc(hidden)]
+    unsafe fn submit(&mut self, submit_infos: &[QueueSubmit<B>], fence: Option<&mut <B::Resources as Resources>::Fence>,
```
Comment (kvark) [inline src/core/src/lib.rs]: why is it unsafe btw?

Comment (msiglreith) [inline src/core/src/lib.rs]: I marked it as `unsafe` as it's not supposed to be called. You could for example submit `GraphicsCommandBuffers` to a `ComputeQueue`, which will probably result in some driver error.

File: examples/cube/main.rs (hunk 1 above)
Comment (msiglreith) [inline examples/cube/main.rs]: noted!

**Pull request merged**